### PR TITLE
logRoles now logs its output to console.

### PR DESCRIPTION
**Pull request by michaellasky**
API is unchanged, logRoles will still return the same string, but now
also logs to the console.

https://github.com/testing-library/dom-testing-library/issues/311

Initial fix for Issue #311 

See discussion in #311 

logRoles still returns a string, but now also logs its output to the console.

**Checklist**:

- [ ] Documentation added to the
      [docs site](https://github.com/alexkrolick/testing-library-docs)
- [ ] Typescript definitions updated
- [ ] Tests
- [ ] Ready to be merged

### CODE REVIEW

**Comment by michaellasky** [comment]
This build is going to fail because of code coverage.  As mentioned in #311, I'm not sure the best way to go about testing a console log side effect with void return value.

File: src/__tests__/role-helpers.js [hunk 1]
```diff
@@ -132,15 +131,6 @@ test('getRoles returns expected roles for various dom nodes', () => {
   })
 })
 
-test('logRoles logs expected roles for various dom nodes', () => {
-  expect.addSnapshotSerializer(jestSerializerAnsi)
```
Comment (kentcdodds) [inline src/__tests__/role-helpers.js]: I missed this in the previous review, but you could add this to https://github.com/testing-library/dom-testing-library/blob/master/tests/setup-env.js instead and then it'll be applied to all snapshots which is probably a good thing.

File: src/__tests__/role-helpers.js [hunk 2]
```diff
@@ -132,15 +131,6 @@ test('getRoles returns expected roles for various dom nodes', () => {
   })
 })
 
-test('logRoles logs expected roles for various dom nodes', () => {
```
Comment (kentcdodds) [inline src/__tests__/role-helpers.js]: To test calls to `console.log` you could do something like this:

```javascript
jest.spyOn(console, 'log').mockImplementationOnce(() => {})

const {section} = setup()
logRoles(section)

expect(console.log).toHaveBeenCalledTimes(1)
expect(console.log.mock.calls[0][0]).toMatchSnapshot()

console.log.mockRestore()
```

Comment (kentcdodds) [reply src/__tests__/role-helpers.js]: Nice! Looks like you figured it out without my help. Two things:

1. Let's move that `addSnapshotSerializer` stuff to the `setup-env.js` file.
2. Make sure to call `mockRestore()`

File: src/__tests__/role-helpers.js [hunk 3]
```diff
@@ -132,13 +132,16 @@ test('getRoles returns expected roles for various dom nodes', () => {
...
+test('logRoles calls console.log with output from prettyRoles', () => {
   const {section} = setup()
-  const output = logRoles(section)
+  expect.addSnapshotSerializer(jestSerializerAnsi)
 
-  expect(output).toMatchSnapshot()
+  jest.spyOn(console, 'log')
+  logRoles(section)
+  // eslint-disable-next-line no-console
+  expect(console.log.mock.calls).toHaveLength(1)
```
Comment (kentcdodds) [inline src/__tests__/role-helpers.js]: This should probably be swapped with `expect(console.log).toHaveBeenCalledTimes(1)` to improve any error message that may occur.

File: src/__tests__/role-helpers.js (hunk 1 above)
Comment (michaellasky) [reply src/__tests__/role-helpers.js]: Will do!  Test is fixed, was overthinking things.. been a long day. 

Adding the serializer to setup-env incoming

File: src/__tests__/role-helpers.js (hunk 3 above)
Comment (michaellasky) [reply src/__tests__/role-helpers.js]: Done [5fefa1](https://github.com/testing-library/dom-testing-library/pull/312/commits/5fefa10d167518fa396538a09ef428b81f2ce45d)

File: src/__tests__/role-helpers.js (hunk 2 above)
Comment (michaellasky) [reply src/__tests__/role-helpers.js]: Handled [5fefa1](https://github.com/testing-library/dom-testing-library/pull/312/commits/5fefa10d167518fa396538a09ef428b81f2ce45d)

**Pull request merged**